wave_utils/include/wave/utils/world_frame_conversions.hpp

```diff
@@ -0,0 +1,133 @@
+/* Copyright (c) 2017, Waterloo Autonomous Vehicles Laboratory (WAVELab),
+ * Waterloo Intelligent Systems Engineering Lab (WISELab),
+ * University of Waterloo.
+ *
+ * Refer to the accompanying LICENSE file for license information.
+ *
+ * ############################################################################
+ ******************************************************************************
+ |                                                                            |
+ |                         /\/\__/\_/\      /\_/\__/\/\                       |
+ |                         \          \____/          /                       |
+ |                          '----________________----'                        |
+ |                              /                \                            |
+ |                            O/_____/_______/____\O                          |
+ |                            /____________________\                          |
+ |                           /    (#UNIVERSITY#)    \                         |
+ |                           |[**](#OFWATERLOO#)[**]|                         |
+ |                           \______________________/                         |
+ |                            |_""__|_,----,_|__""_|                          |
+ |                            ! !                ! !                          |
+ |                            '-'                '-'                          |
+ |       __    _   _  _____  ___  __  _  ___  _    _  ___  ___   ____  ____   |
+ |      /  \  | | | ||_   _|/ _ \|  \| |/ _ \| \  / |/ _ \/ _ \ /     |       |
+ |     / /\ \ | |_| |  | |  ||_||| |\  |||_|||  \/  |||_||||_|| \===\ |====   |
+ |    /_/  \_\|_____|  |_|  \___/|_| \_|\___/|_|\/|_|\___/\___/ ____/ |____   |
+ |                                                                            |
+ ******************************************************************************
+ * ############################################################################
+ *
+ * File: world_frame_conversions.hpp
+ * Desc: Header file for world frame conversion functions
+ * Auth: Michael Smart <michael.smart@uwaterloo.ca>
+ *
+ * ############################################################################
+*/
+
+#ifndef WAVE_SPATIAL_UTILS_WORLD_FRAME_CONVERSIONS_HPP
+#define WAVE_SPATIAL_UTILS_WORLD_FRAME_CONVERSIONS_HPP
+
+#include <cmath>
+#include <GeographicLib/Geocentric.hpp>
+#include <GeographicLib/LocalCartesian.hpp>
+
+namespace wave {
+
+/** Converts a point from LLH (Latitude [deg], Longitude [deg], Height[m]) to
+ *  ECEF.
+ *
+ *  @param[in] llh the input llh point as (Latitude, Longitude, Height). Height
+ *  is relative to the WGS84 ellipsoid.
+ *  @param[out] ecef the corresponding point in the geocentric ECEF frame.
+ */
+void ecefPointFromLLH(const double llh[3], double ecef[3]);
```

Comment:
So one thing I do want to do in the future is move away from these arrays...we could use Eigen vectors here. It's not as big of a deal for this one, but more so for the [4][4] arrays listed below, when we could use Eigen's `Affine3d` class.